src/module.c

```diff
@@ -6461,6 +6461,49 @@ const char *RM_CallReplyProto(RedisModuleCallReply *reply, size_t *len) {
     return callReplyGetProto(reply, len);
 }
 
+/* Redis only exposes "side effects" of command execution at the end of an execution unit.
+ *
+ * Side effects currently handled are
+ * - propagating replication/AOF operations
+ * - executing the commands of clients blocked on keys modified by commands
+ *
+ * In general, for modules, execution unit start and end with the taking/release of the GIL.  For module code executed
```

Comment:
We ended up calling the notification job `AddPostNotificaitonJob`. There is no concept of execution unit in modules today. I think we need to spend more effort here explaining what an execution unit is.